src/serverlist.cpp

```diff
@@ -469,6 +519,7 @@ void CServerListManager::CentralServerQueryServerList ( const CHostAddress& Inet
                     // servers behind a NAT and dealing with external, public
                     // clients.
                     vecServerInfo[iIdx].HostAddr = ServerList[iIdx].LHostAddr;
+                    // ?? Shouldn't this send the ping, as below ??
```

Comment:
@hoffie ?

Reply:
Hrm, so this is about making the `--serverpublicip` Directory-side logic more consistent with the regular code path regarding NAT traversal, right?

I agree that a ping should be send to be as close to the normal path.

On the other hand, this feature is only used by people who already have the capability to set up proper port forwardings (otherwise they couldn't run a public directory server behind NAT). Therefore, I'd argue that it might not be necessary. As this hasn't been reported as a bug or missing feature either, I'd rather not do something right now. If anyone disagrees, please say so and I can open a PR.